### PR TITLE
New OSS::Carousel component

### PR DESCRIPTION
**Pull request by Miexil**
### What does this PR do?
New OSS::Carousel component

```
interface OSSCarouselArgs {
  buttonIcon?: string;
  animationStyle?: 'shift' | 'slide';
  showIndicators?: boolean;
  showControls?: 'overlay' | 'outside';
  autoPlay?: number;
}
```

Usage:
```handlebars
  <OSS::Carousel @showIndicators={{this.showIndicators}} @showControls={{this.showControls}}
                 @animationStyle={{this.animationStyle}} @buttonIcon={{this.buttonIcon}}
                 @autoPlay={{this.autoPlay}}>
    <:pages>
      <div class="page">
        <OSS::Banner @icon="fas fa-image" @title="PAGE 1"/>
      </div>
      <div class="page">
        <OSS::Banner @icon="fas fa-image" @title="PAGE 2"/>
      </div>
      <div class="page">
        <OSS::Banner @icon="fas fa-image" @title="PAGE 1"/>
      </div>
    </:pages>
  </OSS::Carousel>
```

## Examples

With controls, `outside`:
Screenshot 2024-10-03 at 14 59 18

With controls, `overlay` (would be more useful with pictures for example):
Screenshot 2024-10-03 at 14 59 54

Without `indicators`:
Screenshot 2024-10-03 at 15 00 39

With custom `fas fa-robot` `indicators`:
Screenshot 2024-10-03 at 15 01 22

## Video of animations

With default `shift` animation:
https://github.com/user-attachments/assets/645cf53a-ebc5-4558-8b70-95a639ef19b2

With `slide` animation:
https://github.com/user-attachments/assets/43372e49-f23d-4cfe-a20c-45bdbaea5d64

### Good PR checklist
- [x] Title makes sense
- [x] Is against the correct branch
- [x] Only addresses one issue
- [x] Properly assigned
- [x] Added/updated tests
- [x] Added/updated documentation
- [x] Properly labeled

### CODE REVIEW

File: addon/components/o-s-s/carousel.ts [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+  }
+
+  @action
+  displayPage(page: HTMLElement): void {
+    if (this.pages.indexOf(page) === this.currentPageIndex || this.ongoingAnimation) {
+      return;
+    }
+    this.ongoingAnimation = true;
+
+    if (this.animationStyle === 'slide') {
```
Comment (OwenCoogan) [inline addon/components/o-s-s/carousel.ts:86]: Not sure here but can you make a private to manage & call the slide animation handler ?

File: app/styles/organisms/carousel.less [hunk 2]
```diff
@@ -0,0 +1,167 @@
+.oss-carousel {
+  display: flex;
+  flex: 1;
+  flex-direction: column;
+  align-items: center;
+  gap: var(--spacing-px-12);
+  position: relative;
+
+  .page-btn {
+    padding: 3px;
```
Comment (OwenCoogan) [inline app/styles/organisms/carousel.less:10]: you can use var(--spacing-px-3) here

File: app/styles/organisms/carousel.less [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+  .carousel-control {
+    position: absolute;
+    top: 0;
+    bottom: 0;
+    z-index: 1;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    width: 24px;
```
Comment (OwenCoogan) [inline app/styles/organisms/carousel.less:28]: var(--spacing-px-24) here too :)

**Review by OwenCoogan — APPROVED** [review]
Pre approved with a few nits

File: addon/components/o-s-s/carousel.ts [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    this.triggerAnimation(page);
+    page.classList.add('page--active');
+    this.prevPageIndex = this.currentPageIndex;
+    this.currentPageIndex = this.pages.indexOf(page);
+  }
+
+  willDestroy() {
+    super.willDestroy();
+    clearInterval(this.autoPlayInterval);
+  }
```
Comment (JulienVannier66) [inline addon/components/o-s-s/carousel.ts:94]: Not used, right?
I didn't see any `will-destroy` in the template. If you keep it, don't forget the return type

File: app/styles/organisms/carousel.less [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+    bottom: 0;
+    z-index: 1;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    width: var(--spacing-px-24);
+    padding: 0;
+    color: var(--color-gray-500);
+    text-align: center;
+    transition: opacity .15s ease;
```
Comment (JulienVannier66) [inline app/styles/organisms/carousel.less:32]: 300ms is too long?

File: app/styles/organisms/carousel.less [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+    overflow: hidden;
+    width: 100%;
+
+    .page {
+      width: 100%;
+      margin-right: -100%;
+      display: none;
+      float: left;
+      position: relative;
+      transition: transform .5s ease-in-out;
```
Comment (JulienVannier66) [inline app/styles/organisms/carousel.less:56]: Sync with design for this duration of animation?

File: addon/components/o-s-s/carousel.ts (hunk 4 above)
Comment (Miexil) [reply addon/components/o-s-s/carousel.ts:94]: This is from Ember's component hook cycle, it is triggered when the component is destroyed :) 
Adding the return type though

File: app/styles/organisms/carousel.less (hunk 5 above)
Comment (Miexil) [reply app/styles/organisms/carousel.less:32]: Adjusted to be 500ms like the others

File: app/styles/organisms/carousel.less (hunk 6 above)
Comment (Miexil) [reply app/styles/organisms/carousel.less:56]: Nope, I replicated bootstrap's timings for this one as we will need to replace existing carousels in the software.

File: addon/components/o-s-s/carousel.hbs [hunk 7]
```diff
@@ -0,0 +1,22 @@
+<div class="oss-carousel" {{did-insert this.initialize}} ...attributes>
+  {{#if this.showIndicators}}
+    <div class="fx-row">
+      {{#each this.pages as |page index|}}
```
Comment (aprentout) [inline addon/components/o-s-s/carousel.hbs:4]: Nit: as `page` is not used elsewhere shouldn't we set `_` instead ?

**Review by aprentout — APPROVED** [review]

**Review by aprentout — APPROVED** [review]

**Pull request merged**